src/quark-tools.ts

```diff
@@ -0,0 +1,386 @@
+            parentdecorationsArray.push(methodSegmentDecoration);
+
+            const parentdecorationType = vscode.window.createTextEditorDecorationType(
+                {
+                    isWholeLine: true,
+                    dark: {
+                        backgroundColor: "#3c5707",
+                    },
+                    light: {
+                        backgroundColor: "#dcfddc",
+                    },
+                }
+            );
+            const apidecorationType = vscode.window.createTextEditorDecorationType(
+                {
+                    fontWeight: "bold",
+                    isWholeLine: true,
+                    dark: {
+                        backgroundColor: "#5b2334",
+                    },
+                    light: {
+                        backgroundColor: "#ffc2c3",
+                    },
+                }
+            );
+
+            e.setDecorations(parentdecorationType, parentdecorationsArray);
+            e.setDecorations(apidecorationType, apidecorationsArray);
+
+            e.selection = new vscode.Selection(
+                parentFunctionPosition,
+                parentFunctionPosition
+            );
+
+            vscode.commands.executeCommand("revealLine", {
+                lineNumber: mdSegment[0],
+                at: "top",
+            });
+        });
+    });
+}
+
+export namespace Quark {
+    /**
+     * Check if Quark are installed
+     * @return if quark installed or not
+     */
+    export function checkQuarkInstalled(): boolean {
+        // TODO: Find a better way to check python and quark install.
+
+        outputChannel.appendLine(`Check if ${quarkDataDirPath} exist...`);
+        if (!fs.existsSync(quarkDataDirPath)) {
+            outputChannel.appendLine(`${quarkDataDirPath} not exist.`);
+            outputChannel.appendLine(`mkdir ${quarkDataDirPath}`);
+            fs.mkdirSync(quarkDataDirPath);
+        }
+        const cmd: string = process.platform.startsWith("win")
+            ? `quark`
+            : `cd ${String(quarkDataDirPath)} && quark`;
+
+        outputChannel.appendLine(`exec: ${cmd}`);
+
+        try {
+            child_process.execSync(cmd);
+            return true;
+        } catch (error) {
+            outputChannel.appendLine(`Catched error from Quark install check`);
+            outputChannel.append(String(error));
+            return false;
+        }
+    }
+
+    /**
+     * Analyzing APK using Quark-Engine.
+     * @param apkFilePath The path of APK file.
+     * @param apkDecodedDir The path of the decoded directory.
+     */
+    export async function analyzeAPK(
+        apkFilePath: string,
+        apkDecodedDir: string
+    ): Promise<void> {
+        const reportPath = path.join(apkDecodedDir, `quarkReport.json`);
+        const quarkRulePath = path.join(quarkDataDirPath, "quark-rules/");
+
+        const script = `cd ${quarkDataDirPath} && quark -a ${apkFilePath} -r ${quarkRulePath} -o ${reportPath}`;
```

Comment:
Same here. Since logs and rules will go in `~/.quark-engine` by default we don't need to do `cd` or to pass `-r`.
```suggestion
        const script = `quark -a ${apkFilePath} -o ${reportPath}`;
```